metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,102 @@
+package prometheus
```

Comment:
I know we already had a discussion and on second thought I don't agree with the sentiment that this should not go in the respective packages. The declaration of the structs, the default nop implementation and the general ownership of the metrics are already there. We import prometheus anyway in Tendermint and as long as we don't have pressurign use-cases for single packages which require prometheus not to be imported I think we should not make decisions based on it. Even if we decide we want "cleaner" imports we can always make sub-packages with the metric providers. As it stands right now it feels dislocated and is harder to maintain as there are multiple places to keep track, which is unnecessary burden.